diff -Nru vdr-plugin-externalplayer-0.1.0/Makefile vdr-plugin-externalplayer-0.1.0.patched/Makefile
--- vdr-plugin-externalplayer-0.1.0/Makefile	2005-10-07 09:36:26.000000000 +0200
+++ vdr-plugin-externalplayer-0.1.0.patched/Makefile	2010-01-02 19:50:58.070430509 +0100
@@ -46,7 +46,7 @@
 
 ### The object files (add further files here):
 
-OBJS = externalplayer.o externalplayer-config.o externalplayer-control.o  externalplayer-i18n.o externalplayer-player.o externalplayer-remotes.o
+OBJS = externalplayer.o externalplayer-config.o externalplayer-control.o  externalplayer-i18n.o externalplayer-player.o 
 
 ### Implicit rules:
 
diff -Nru vdr-plugin-externalplayer-0.1.0/externalplayer-control.c vdr-plugin-externalplayer-0.1.0.patched/externalplayer-control.c
--- vdr-plugin-externalplayer-0.1.0/externalplayer-control.c	2005-10-09 00:14:18.000000000 +0200
+++ vdr-plugin-externalplayer-0.1.0.patched/externalplayer-control.c	2010-01-03 10:16:57.494430833 +0100
@@ -8,6 +8,7 @@
 
 #include "externalplayer-control.h"
 #include "externalplayer-player.h"
+#include <vdr/remote.h>
 
 cStatusExternalplayer::cStatusExternalplayer(sPlayerArgs * nConfig) : cStatus() {
   config = nConfig;
@@ -26,6 +27,10 @@
     : cControl(player = new cPlayerExternalplayer(nConfig->playMode, nConfig, fdsPipe[0])) {
   config = nConfig;
   status = new cStatusExternalplayer(config);
+  if(config->deactivateRemotes) {
+     cRemote::SetEnabled(false);
+     isyslog("externalplayer-plugin: disable remotes");
+  }
   fdWritePipe = fdsPipe[1];
 }
 
@@ -33,6 +38,9 @@
   isyslog("externalplayer-plugin: shutting down player");
   delete player;
   delete status;
+  cRemote::Clear();
+  cRemote::SetEnabled(true);
+  isyslog("externalplayer-plugin: enable remotes");
   player = NULL;
 }
 
diff -Nru vdr-plugin-externalplayer-0.1.0/externalplayer-player.c vdr-plugin-externalplayer-0.1.0.patched/externalplayer-player.c
--- vdr-plugin-externalplayer-0.1.0/externalplayer-player.c	2005-10-08 17:32:37.000000000 +0200
+++ vdr-plugin-externalplayer-0.1.0.patched/externalplayer-player.c	2010-01-02 19:21:26.970928890 +0100
@@ -27,29 +27,15 @@
 cPlayerExternalplayer::cPlayerExternalplayer(ePlayMode playMode, sPlayerArgs * nConfig, int nFdReadPipe)
     : cPlayer(playMode) {
   config = nConfig;
-
   fdReadPipe = nFdReadPipe;
-
-  if (config->deactivateRemotes) {
-    remotesDisable = new cRemotesDisable();
-  }
-  else {
-    remotesDisable = NULL;
-  }
-
 }
 
 cPlayerExternalplayer::~cPlayerExternalplayer() {
   Activate(false);
-  delete remotesDisable;
 }
 
 void cPlayerExternalplayer::Activate(bool On) {
   if (On) {
-    if (remotesDisable != NULL) {
-      remotesDisable->deactivateRemotes();
-    }
-
     int nPid = fork();
     if (nPid == 0) {
       if (config->slaveMode) {
@@ -67,10 +53,6 @@
     }
   }
   else {
-    if (remotesDisable != NULL) {
-      remotesDisable->reactivateRemotes();
-    }
-
     if (pid != 0) {
       if (kill(pid, 0) == 0) {
         kill(pid, SIGTERM);
diff -Nru vdr-plugin-externalplayer-0.1.0/externalplayer-player.h vdr-plugin-externalplayer-0.1.0.patched/externalplayer-player.h
--- vdr-plugin-externalplayer-0.1.0/externalplayer-player.h	2005-10-06 20:55:37.000000000 +0200
+++ vdr-plugin-externalplayer-0.1.0.patched/externalplayer-player.h	2010-01-02 17:43:21.926428962 +0100
@@ -11,14 +11,12 @@
 
 #include <vdr/player.h>
 
-#include "externalplayer-remotes.h"
 #include "externalplayer-config.h"
 
 using namespace std;
 
 class cPlayerExternalplayer : public cPlayer {
 private:
-  cRemotesDisable * remotesDisable;
   sPlayerArgs * config;
   int fdReadPipe;
 protected:
diff -Nru vdr-plugin-externalplayer-0.1.0/externalplayer-remotes.c vdr-plugin-externalplayer-0.1.0.patched/externalplayer-remotes.c
--- vdr-plugin-externalplayer-0.1.0/externalplayer-remotes.c	2005-10-06 20:55:37.000000000 +0200
+++ vdr-plugin-externalplayer-0.1.0.patched/externalplayer-remotes.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,60 +0,0 @@
-/*
- * externalplayer-remotes.c: A plugin for the Video Disk Recorder
- *
- * See the README file for copyright information and how to reach the author.
- *
- * $Id$
- */
-
-#include <string.h>
-
-#include "externalplayer-remotes.h"
-
-void cRemotesDisable::deactivateRemotes() {
-  if (!deactivated) {
-    isyslog("externalplayer-plugin: deactivating remotes");
-    for (cRemote * i = Remotes.First(); i != NULL; i = Remotes.Next(i)) {
-      if (strcmp(i->Name(), "LIRC") == 0) {
-        ((cRemotesDisableHelper *) i)->deactivate();
-      }
-      else if (strcmp(i->Name(), "RCU") == 0) {
-        ((cRemotesDisableHelper *) i)->deactivate();
-      }
-    }
-  }
-
-  deactivated = true;
-}
-
-void cRemotesDisable::reactivateRemotes() {
-  if (deactivated) {
-    isyslog("externalplayer-plugin: reactivating remotes");
-    for (cRemote * i = Remotes.First(); i != NULL; i = Remotes.Next(i)) {
-      if (strcmp(i->Name(), "LIRC") == 0) {
-        ((cRemotesDisableHelper *) i)->reactivate();
-      }
-      else if (strcmp(i->Name(), "RCU") == 0) {
-        ((cRemotesDisableHelper *) i)->reactivate();
-      }
-    }
-  }
-
-  deactivated = false;
-}
-
-// --- cRemotesDisableHelper -------------------------------------------------
-
-cRemotesDisableHelper::cRemotesDisableHelper(const char * name) : cRemote(name) {
-}
-
-void cRemotesDisableHelper::deactivate() {
-  if (Active()) {
-    Cancel();
-  }
-}
-
-void cRemotesDisableHelper::reactivate() {
-  if (!Active()) {
-    Start();
-  }
-}
diff -Nru vdr-plugin-externalplayer-0.1.0/externalplayer-remotes.h vdr-plugin-externalplayer-0.1.0.patched/externalplayer-remotes.h
--- vdr-plugin-externalplayer-0.1.0/externalplayer-remotes.h	2005-10-06 20:55:37.000000000 +0200
+++ vdr-plugin-externalplayer-0.1.0.patched/externalplayer-remotes.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,32 +0,0 @@
-/*
- * externalplayer-remotes.h: A plugin for the Video Disk Recorder
- *
- * See the README file for copyright information and how to reach the author.
- *
- * $Id$
- */
-
-#ifndef _EXTERNALPLAYER_REMOTES_H_
-#define _EXTERNALPLAYER_REMOTES_H_
-
-#include <vdr/thread.h>
-#include <vdr/remote.h>
-
-using namespace std;
-
-class cRemotesDisable {
-private:
-  bool deactivated;
-public:
-  void deactivateRemotes();
-  void reactivateRemotes();
-};
-
-class cRemotesDisableHelper : public cRemote, public cThread {
-public:
-  cRemotesDisableHelper(const char * name);
-  void deactivate();
-  void reactivate();
-};
-
-#endif /*_EXTERNALPLAYER_REMOTES_H_*/
diff -Nru vdr-plugin-externalplayer-0.1.0/externalplayer.c vdr-plugin-externalplayer-0.1.0.patched/externalplayer.c
--- vdr-plugin-externalplayer-0.1.0/externalplayer.c	2005-10-08 22:21:37.000000000 +0200
+++ vdr-plugin-externalplayer-0.1.0.patched/externalplayer.c	2010-01-02 17:43:48.694431611 +0100
@@ -15,7 +15,6 @@
 #include "externalplayer-i18n.h"
 #include "externalplayer-player.h"
 #include "externalplayer-control.h"
-#include "externalplayer-remotes.h"
 
 cPluginExternalplayer::cPluginExternalplayer() {
   playerConfig = NULL;
