diff --git a/src/netceiver.c b/src/netceiver.c
index 4357a31..cf51541 100644
--- a/src/netceiver.c
+++ b/src/netceiver.c
@@ -409,6 +409,10 @@ void find_netcv_adapter(adapter **a)
 	if (recv_init(opts.netcv_if, 23000))
 		LOG("Netceiver init failed");
 
+	/* Call api_sock_init to initialize the socket needed for netceiver tools */
+	if (api_sock_init(API_SOCK_NAMESPACE))
+		LOG("Netceiver API socket init failed");
+
 	sprintf(dbuf, "REEL: Search for %d Netceiver%s on %s... ",
 			opts.netcv_count, opts.netcv_count == 1 ? "" : "s", opts.netcv_if);
 	n = 0;
diff --git a/src/netceiver.h b/src/netceiver.h
index a3d10ee..259148d 100644
--- a/src/netceiver.h
+++ b/src/netceiver.h
@@ -3,6 +3,7 @@
 
 #include "adapter.h"
 
+#define API_SOCK
 #ifndef DISABLE_LINUXDVB
 #undef AOT_CA_PMT
 #include "headers.h"
diff --git a/src/netceiver_mcli_defs.h b/src/netceiver_mcli_defs.h
index 66fd527..8fb1680 100644
--- a/src/netceiver_mcli_defs.h
+++ b/src/netceiver_mcli_defs.h
@@ -197,6 +197,7 @@ typedef struct recv_festatus
 #include "mcast.h"
 #include "recv_ccpp.h"
 #include "recv_tv.h"
+#include "api_server.h"
 #include "tca_handler.h"
 
 #endif
