diff -r cfa4b4faab67 linux/drivers/media/common/saa716x/Makefile
--- a/linux/drivers/media/common/saa716x/Makefile	Sun Feb 24 14:01:09 2013 +0100
+++ b/linux/drivers/media/common/saa716x/Makefile	Fri Jul 26 13:38:44 2013 +0200
@@ -24,3 +24,4 @@
 obj-$(CONFIG_DVB_SAA716X_FF)	  += saa716x_ff.o
 
 EXTRA_CFLAGS = -Idrivers/media/dvb/dvb-core/ -Idrivers/media/dvb/frontends/ -Idrivers/media/dvb-core/ -Idrivers/media/dvb-frontends/
+ccflags-y += -Idrivers/media/tuners/
diff -r cfa4b4faab67 linux/drivers/media/common/saa716x/saa716x_ff_cmd.h
--- a/linux/drivers/media/common/saa716x/saa716x_ff_cmd.h	Sun Feb 24 14:01:09 2013 +0100
+++ b/linux/drivers/media/common/saa716x/saa716x_ff_cmd.h	Fri Jul 26 13:38:44 2013 +0200
@@ -1,6 +1,24 @@
 #ifndef __SAA716x_FF_CMD_H
 #define __SAA716x_FF_CMD_H
 
+#if !defined OSD_RAW_CMD
+typedef struct osd_raw_cmd_s {
+    const void *cmd_data;
+    int cmd_len;
+    void *result_data;
+    int result_len;
+} osd_raw_cmd_t;
+
+typedef struct osd_raw_data_s {
+    const void *data_buffer;
+    int data_length;
+    int data_handle;
+} osd_raw_data_t;
+
+#define OSD_RAW_CMD            _IOWR('o', 162, osd_raw_cmd_t)
+#define OSD_RAW_DATA           _IOWR('o', 163, osd_raw_data_t)
+#endif
+
 extern int sti7109_cmd_init(struct sti7109_dev *sti7109);
 extern int sti7109_raw_cmd(struct sti7109_dev * sti7109,
 			   osd_raw_cmd_t * cmd);
diff -r cfa4b4faab67 linux/drivers/media/common/saa716x/saa716x_i2c.c
--- a/linux/drivers/media/common/saa716x/saa716x_i2c.c	Sun Feb 24 14:01:09 2013 +0100
+++ b/linux/drivers/media/common/saa716x/saa716x_i2c.c	Fri Jul 26 13:38:44 2013 +0200
@@ -721,7 +721,6 @@
 		saa716x_i2c_hwdeinit(i2c, SAA716x_I2C_BUS(i));
 		dprintk(SAA716x_DEBUG, 1, "Removing adapter (%d) %s", i, adapter->name);
 
-		err = i2c_del_adapter(adapter);
 		if (err < 0) {
 			dprintk(SAA716x_ERROR, 1, "Adapter (%d) %s remove failed", i, adapter->name);
 			goto exit;
