diff --git a/plugin/setup.cpp b/plugin/setup.cpp
index 8835735..fa90018 100644
--- a/plugin/setup.cpp
+++ b/plugin/setup.cpp
@@ -146,7 +146,8 @@ void cSetupMarkAd::Store(void)
     setup->SaveInfo=saveinfo;
 }
 
-#define CHNUMWIDTH (numdigits(Channels.MaxNumber())+1)
+LOCK_CHANNELS_READ;
+#define CHNUMWIDTH (numdigits(Channels->MaxNumber())+1)
 
 cSetupMarkAdList::cSetupMarkAdList(struct setup *Setup)
         :cOsdMenu("",CHNUMWIDTH)
@@ -168,7 +169,8 @@ cSetupMarkAdList::cSetupMarkAdList(struct setup *Setup)
                 char *m=strchr(name,'-');
                 if (m) *m=0;
 
-                for (cChannel *channel = Channels.First(); channel; channel = Channels.Next(channel))
+                LOCK_CHANNELS_READ;
+                for (const cChannel *channel = Channels->First(); channel; channel = Channels->Next(channel))
                 {
                     if (channel->Name())
                     {
diff --git a/plugin/status.cpp b/plugin/status.cpp
index a691477..889940d 100644
--- a/plugin/status.cpp
+++ b/plugin/status.cpp
@@ -139,8 +139,9 @@ void cStatusMarkAd::TimerChange(const cTimer *Timer, eTimerChange Change)
 bool cStatusMarkAd::LogoExists(const cDevice *Device,const char *FileName)
 {
     if (!FileName) return false;
-    cTimer *timer=NULL;
-    for (cTimer *Timer = Timers.First(); Timer; Timer=Timers.Next(Timer))
+    const cTimer *timer=NULL;
+    LOCK_TIMERS_READ;
+    for (const cTimer *Timer = Timers->First(); Timer; Timer=Timers->Next(Timer))
     {
 #if APIVERSNUM>=10722
         if (Timer->Recording() && const_cast<cDevice *>(Device)->IsTunedToTransponder(Timer->Channel()) &&
