diff --git a/handler.h b/handler.h
index ed80e42..6e57e96 100644
--- a/handler.h
+++ b/handler.h
@@ -920,7 +920,8 @@ class cEpg2VdrEpgHandler : public cEpgHandler
             int merge = mapDb->getIntValue(cTableChannelMap::fiMerge);
 
             const char* strChannelId = mapDb->getStrValue(cTableChannelMap::fiChannelId);
-            cChannel* channel = Channels.GetByChannelID(tChannelID::FromString(strChannelId));
+            LOCK_CHANNELS_READ;
+            const cChannel* channel = Channels->GetByChannelID(tChannelID::FromString(strChannelId));
 
             // update channelname in channelmap
             
diff --git a/timer.c b/timer.c
index 4aed3d5..4440548 100644
--- a/timer.c
+++ b/timer.c
@@ -30,7 +30,8 @@ int cUpdate::updateTimer()
    {
       int exist = no;
 
-      for (cTimer* t = Timers.First(); t; t = Timers.Next(t)) 
+      LOCK_TIMERS_READ;
+      for (const cTimer* t = Timers->First(); t; t = Timers->Next(t)) 
       {
          if (t->StartTime() == timerDb->getIntValue(cTableTimers::fiStartTime) &&
              strcmp(t->Channel()->GetChannelID().ToString(), timerDb->getStrValue(cTableTimers::fiChannelId)) == 0)
@@ -49,7 +50,8 @@ int cUpdate::updateTimer()
 
    // update timers
 
-   for (cTimer* t = Timers.First(); t; t = Timers.Next(t)) 
+   LOCK_TIMERS_READ;
+   for (const cTimer* t = Timers->First(); t; t = Timers->Next(t)) 
    {
       int insert;
       
diff --git a/update.c b/update.c
index 85ae1bd..cdccf5f 100644
--- a/update.c
+++ b/update.c
@@ -462,9 +462,10 @@ int cUpdate::exitDb()
 
 cTimer* cUpdate::getTimerOf(const cEvent* event) const
 {
-   for (cTimer* t = Timers.First(); t; t = Timers.Next(t)) 
+   LOCK_TIMERS_READ;
+   for (const cTimer* t = Timers->First(); t; t = Timers->Next(t)) 
       if (t->Event() == event)
-         return t;
+         return (cTimer*) t;
 
    return 0;
 }
@@ -794,9 +795,6 @@ int cUpdate::refreshEpg()
    {
       tell(1, "Removing all events from epg");
 
-      while (!cSchedules::ClearAll())
-         tell(0, "Warning: Clear EPG failed, can't get lock. Retrying ...");
-
       lastUpdateAt = 0;
    }
 
@@ -830,14 +828,7 @@ int cUpdate::refreshEpg()
 
       // lock holen
 
-      if (!(ss = (cSchedules*)cSchedules::Schedules(*schedulesLock)))
-      {
-         delete schedulesLock;
-         tell(0, "Error, can't get lock on schedules, aborting refresh!");
-         selectAllChannels->freeResult();
-         
-         return fail;  // #TODO -> break!?
-      }
+      LOCK_SCHEDULES_READ;
 
       // get schedule (channel)
       
@@ -896,7 +887,7 @@ int cUpdate::refreshEpg()
       // Kanal fertig machen ..
 
       s->Sort();
-      ss->SetModified(s);
+      s->SetModified();
       
       // lock freigeben 
       
