diff -ruN media_build-bst.orig/linux/drivers/media/pci/cx88/cx88-dvb.c media_build-bst/linux/drivers/media/pci/cx88/cx88-dvb.c
--- media_build-bst.orig/linux/drivers/media/pci/cx88/cx88-dvb.c	2013-01-31 03:42:51.000000000 +0100
+++ media_build-bst/linux/drivers/media/pci/cx88/cx88-dvb.c	2013-08-05 01:24:46.677924998 +0200
@@ -1092,7 +1092,7 @@
 			if (!dvb_attach(isl6421_attach,
 					fe0->dvb.frontend,
 					&dev->core->i2c_adap,
-					0x08, ISL6421_DCL, 0x00))
+					0x08, ISL6421_DCL, 0x00, false))
 				goto frontend_detach;
 		}
 		/* MFE frontend 2 */
@@ -1330,7 +1330,7 @@
 					       &core->i2c_adap);
 		if (fe0->dvb.frontend) {
 			if (!dvb_attach(isl6421_attach, fe0->dvb.frontend,
-					&core->i2c_adap, 0x08, ISL6421_DCL, 0x00))
+					&core->i2c_adap, 0x08, ISL6421_DCL, 0x00, false))
 				goto frontend_detach;
 		}
 		break;
@@ -1453,7 +1453,7 @@
 			if (!dvb_attach(isl6421_attach,
 					fe0->dvb.frontend,
 					&dev->core->i2c_adap,
-					0x08, ISL6421_DCL, 0x00))
+					0x08, ISL6421_DCL, 0x00, false))
 				goto frontend_detach;
 		}
 		/* MFE frontend 2 */
@@ -1481,7 +1481,7 @@
 			if (!dvb_attach(isl6421_attach,
 					fe0->dvb.frontend,
 					&dev->core->i2c_adap,
-					0x08, ISL6421_DCL, 0x00))
+					0x08, ISL6421_DCL, 0x00, false))
 				goto frontend_detach;
 		}
 		break;
