diff -ruN epgsearch-1.0.1.beta2/epgsearchsvdrp.c epgsearch-1.0.1.beta2.patched/epgsearchsvdrp.c
--- epgsearch-1.0.1.beta2/epgsearchsvdrp.c	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/epgsearchsvdrp.c	2012-12-08 13:57:05.000000000 +0100
@@ -560,7 +560,7 @@
             strftime(bufStart, sizeof(bufStart), "%H%M", localtime_r(&start, &tm_r));
             strftime(bufEnd, sizeof(bufEnd), "%H%M", localtime_r(&stop, &tm_r));
 
-            int timerMatch;
+            eTimerMatch timerMatch;
             bool hasTimer = false;
             if (Timers.GetMatch(pEvent, &timerMatch))
                hasTimer = (timerMatch == tmFull);
diff -ruN epgsearch-1.0.1.beta2/mail.c epgsearch-1.0.1.beta2.patched/mail.c
--- epgsearch-1.0.1.beta2/mail.c	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/mail.c	2012-12-08 13:57:40.000000000 +0100
@@ -49,7 +49,7 @@
     const cEvent* pEvent = GetEvent();
     if (!pEvent) return "";
 
-    int TimerMatch = tmNone;
+    eTimerMatch TimerMatch = tmNone;
     cTimer* pTimer = Timers.GetMatch(pEvent, &TimerMatch);
     if (!pTimer) return "";
 
diff -ruN epgsearch-1.0.1.beta2/menu_commands.c epgsearch-1.0.1.beta2.patched/menu_commands.c
--- epgsearch-1.0.1.beta2/menu_commands.c	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/menu_commands.c	2012-12-08 13:58:16.000000000 +0100
@@ -117,7 +117,7 @@
 {
    if (!event) return osContinue;
 
-   int timerMatch = tmNone;
+   eTimerMatch timerMatch = tmNone;
    cTimer* timer = Timers.GetMatch(event, &timerMatch);
    if (timerMatch == tmFull)
    {
diff -ruN epgsearch-1.0.1.beta2/menu_event.c epgsearch-1.0.1.beta2.patched/menu_event.c
--- epgsearch-1.0.1.beta2/menu_event.c	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/menu_event.c	2012-12-08 13:58:48.000000000 +0100
@@ -92,7 +92,7 @@
       cEventObj* eventObjPrev = GetPrev(event);
       cEventObj* eventObjNext = GetNext(event);
 
-      int timerMatch = tmNone;
+      eTimerMatch timerMatch = tmNone;
       Timers.GetMatch(event, &timerMatch);
       const char* szRed = trVDR("Button$Record");
       if (timerMatch == tmFull)
diff -ruN epgsearch-1.0.1.beta2/menu_main.c epgsearch-1.0.1.beta2.patched/menu_main.c
--- epgsearch-1.0.1.beta2/menu_main.c	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/menu_main.c	2012-12-08 13:59:12.000000000 +0100
@@ -181,7 +181,7 @@
   if (item) {
       if (item->timerMatch == tmFull)
       {
-	  int tm = tmNone;
+	  eTimerMatch tm = tmNone;
 	  cTimer *timer = Timers.GetMatch(item->event, &tm);
 	  if (timer)
 	    {
diff -ruN epgsearch-1.0.1.beta2/menu_searchresults.c epgsearch-1.0.1.beta2.patched/menu_searchresults.c
--- epgsearch-1.0.1.beta2/menu_searchresults.c	2012-06-11 18:51:08.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/menu_searchresults.c	2012-12-08 14:06:23.000000000 +0100
@@ -176,7 +176,7 @@
    previewTimer = false;
    episodeOnly = false;
    menuTemplate = NULL;
-   timerMatch = 0;
+   int timerMatch = 0;
    inSwitchList = false;
    event = NULL;
    search = NULL;
@@ -224,7 +224,7 @@
    if (item) {
       if (item->timerMatch == tmFull)
       {
-         int tm = tmNone;
+         eTimerMatch tm = tmNone;
          cTimer *timer = Timers.GetMatch(item->event, &tm);
          if (timer)
 	   {
diff -ruN epgsearch-1.0.1.beta2/menu_searchresults.h epgsearch-1.0.1.beta2.patched/menu_searchresults.h
--- epgsearch-1.0.1.beta2/menu_searchresults.h	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/menu_searchresults.h	2012-12-08 13:59:37.000000000 +0100
@@ -50,7 +50,7 @@
     bool episodeOnly;
     cMenuTemplate* menuTemplate;
  public:
-    int timerMatch;
+    eTimerMatch timerMatch;
     bool inSwitchList;
     const cEvent *event;
     const cSearchExt* search;
diff -ruN epgsearch-1.0.1.beta2/menu_whatson.c epgsearch-1.0.1.beta2.patched/menu_whatson.c
--- epgsearch-1.0.1.beta2/menu_whatson.c	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/menu_whatson.c	2012-12-08 14:10:35.000000000 +0100
@@ -516,7 +516,7 @@
    {
       if (item->timerMatch == tmFull)
       {
-         int tm = tmNone;
+         eTimerMatch tm = tmNone;
          cTimer *timer = Timers.GetMatch(item->event, &tm);
          if (timer)
 	   {
diff -ruN epgsearch-1.0.1.beta2/menu_whatson.h epgsearch-1.0.1.beta2.patched/menu_whatson.h
--- epgsearch-1.0.1.beta2/menu_whatson.h	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/menu_whatson.h	2012-12-08 14:06:57.000000000 +0100
@@ -35,7 +35,7 @@
   const cEvent *event;
   cChannel *channel;
   showMode mode;
-  int timerMatch;
+  eTimerMatch timerMatch;
   bool inSwitchList;
   cMenuTemplate* menuTemplate;
 
diff -ruN epgsearch-1.0.1.beta2/services.c epgsearch-1.0.1.beta2.patched/services.c
--- epgsearch-1.0.1.beta2/services.c	2012-06-11 18:43:52.000000000 +0200
+++ epgsearch-1.0.1.beta2.patched/services.c	2012-12-08 14:11:13.000000000 +0100
@@ -143,7 +143,7 @@
          strftime(bufStart, sizeof(bufStart), "%H%M", localtime_r(&start, &tm_r));
          strftime(bufEnd, sizeof(bufEnd), "%H%M", localtime_r(&stop, &tm_r));
 
-         int timerMatch;
+         eTimerMatch timerMatch;
          bool hasTimer = false;
          if (Timers.GetMatch(pEvent, &timerMatch))
             hasTimer = (timerMatch == tmFull);
