diff -ruN vdr-plugin/Makefile vdr-plugin/Makefile
--- vdr-plugin/Makefile	2010-04-30 21:29:50.000000000 +0200
+++ vdr-plugin.patched/Makefile	2010-05-27 17:45:47.430781093 +0200
@@ -34,6 +34,10 @@
 LIBDIR = ../../lib
 TMPDIR = /tmp
 
+### Make sure that necessary options are included:
+
+-include $(VDRDIR)/Make.global
+
 ### Allow user defined options to overwrite defaults:
 
 -include $(VDRDIR)/Make.config
