diff -ruN suspendoutput-2.0.0.orig/suspendoutput.c suspendoutput-2.0.0/suspendoutput.c
--- suspendoutput-2.0.0.orig/suspendoutput.c	2015-09-14 22:35:58.935178335 +0200
+++ suspendoutput-2.0.0/suspendoutput.c	2015-09-14 22:39:06.814855736 +0200
@@ -361,11 +361,10 @@
 
     if ( !cRecordControls::Active() && 
 #if APIVERSNUM > 20101
-	 !RecordingsHandler.Active() && 
+	 !RecordingsHandler.Active()) {
 #else
-	 !cCutter::Active() && 
+	 !cCutter::Active()) {
 #endif
-	 !Interface->HasSVDRPConnection()) {
 
 #define ACTIVITYTIMEOUT 600   // 10 min
 
