diff -ruN vdr-plugin-dbus2vdr-21.orig/osd.c vdr-plugin-dbus2vdr-21/osd.c
--- vdr-plugin-dbus2vdr-21.orig/osd.c	2014-01-26 16:22:59.000000000 +0100
+++ vdr-plugin-dbus2vdr-21/osd.c	2015-02-11 14:22:20.501140585 +0100
@@ -62,7 +62,7 @@
     int x, y;
     const uint8_t *pixel;
     png::image<png::rgba_pixel> *pngfile;
-    while (cPixmapMemory *pm = RenderPixmaps()) {
+    while (cPixmapMemory *pm = dynamic_cast<cPixmapMemory *>(RenderPixmaps())) {
           /*write = true;*/
           vp = &pm->ViewPort();
           vx = vp->X();
@@ -83,7 +83,7 @@
 
           counter++;
           delete pngfile;
-          delete pm;
+          DestroyPixmap(pm);
           }
   }
 #endif
