diff --git a/si_types.h b/si_types.h
index e20b9bd..d153a87 100644
--- a/si_types.h
+++ b/si_types.h
@@ -62,7 +62,7 @@ typedef struct section_buf {
   time_t running_time;
   struct section_buf * next_seg;        // this is used to handle segmented tables (like NIT-other)
   pList  garbage;
-} section_t, * p_section_t;
+};
 
 /*******************************************************************************
 /* service type.
@@ -114,7 +114,7 @@ struct service {
   uint32_t logical_channel_number;
   uint8_t  running;
   void   * priv;
-} service_t, * p_service_t;
+};
 
 /*******************************************************************************
 /* transponder type.
@@ -201,7 +201,7 @@ struct transponder {
   /*----------------------------*/
   char * network_name;
   network_change_t network_change;
-} __attribute__((packed))  transponder_t, * p_transponder_t;
+} __attribute__((packed));
 
 /*******************************************************************************
 /* satellite channel routing type.
