diff --git a/watch.c b/watch.c
index 6414c29..b9b3693 100644
--- a/watch.c
+++ b/watch.c
@@ -138,7 +138,8 @@ void cVFDWatch::shutdown(int nExitMode) {
   }
 
   if(this->isopen()) {
-    cTimer* t = Timers.GetNextActiveTimer();
+    LOCK_TIMERS_READ;
+    const cTimer* t = Timers->GetNextActiveTimer();
 
     switch(nExitMode) {
       case eOnExitMode_NEXTTIMER:
@@ -822,7 +823,8 @@ void cVFDWatch::Channel(int ChannelNumber)
         chName = NULL;
     }
 
-    cChannel * ch = Channels.GetByNumber(ChannelNumber);
+    LOCK_CHANNELS_READ;
+    const cChannel * ch = Channels->GetByNumber(ChannelNumber);
     if(ch) {
       chID = ch->GetChannelID();
       chPresentTime = 0;
@@ -840,11 +842,10 @@ bool cVFDWatch::Program() {
 
     bool bChanged = false;
     const cEvent * p = NULL;
-    cSchedulesLock schedulesLock;
-    const cSchedules * schedules = cSchedules::Schedules(schedulesLock);
-    if (chID.Valid() && schedules)
+    LOCK_SCHEDULES_READ;
+    if (chID.Valid() && Schedules)
     {
-        const cSchedule * schedule = schedules->GetSchedule(chID);
+        const cSchedule * schedule = Schedules->GetSchedule(chID);
         if (schedule)
         {
             if ((p = schedule->GetPresentEvent()) != NULL)
