diff --git a/Makefile b/Makefile
index d0f256c..94bc77e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,28 +1,16 @@
 #
-# Softcam plugin to VDR
+# Makefile for a Video Disk Recorder plugin
 #
-# This code is free software; you can redistribute it and/or
-# modify it under the terms of the GNU General Public License
-# as published by the Free Software Foundation; either version 2
-# of the License, or (at your option) any later version.
-#
-# This code is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
-# Or, point your browser to http://www.gnu.org/copyleft/gpl.html
+# $Id: Makefile 2.13 2012/12/21 11:36:15 kls Exp $
 
 # The official name of this plugin.
 # This name will be used in the '-P...' option of VDR to load the plugin.
 # By default the main source file also carries this name.
-#
+
+
 PLUGIN = sc
 
-### The version number of this plugin
+### The version number of this plugin (taken from the main source file):
 
 DISTFILE = .distvers
 HGARCHIVE = .hg_archival.txt
@@ -41,80 +29,59 @@ SCAPIVERS := $(shell sed -ne '/define SCAPIVERS/ s/^.[a-zA-Z ]*\([0-9]*\).*$$/\1
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+# Use package data if installed...otherwise assume we're under the VDR source directory:
+PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
+LIBDIR = $(call PKGCFG,libdir)
+LOCDIR = $(call PKGCFG,locdir)
+PLGCFG = $(call PKGCFG,plgcfg)
+#
+TMPDIR ?= /tmp
 SYSDIR = ./systems
-PREDIR = ./systems-pre
-TMPDIR = /tmp
 
-### The C++ compiler and options:
+### The compiler options:
 
-CXX      ?= g++
-CXXFLAGS ?= -O2 -g -Wall -Woverloaded-virtual
+export CFLAGS   = $(call PKGCFG,cflags)
+export CXXFLAGS = $(call PKGCFG,cxxflags)
 
-### Includes and Defines
+### The version number of VDR's plugin API:
 
-INCLUDES      = -I$(VDRDIR)/include
-DEFINES       = -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
-SHAREDDEFINES = -DAPIVERSNUM=$(APIVERSNUM) -D_GNU_SOURCE
-LIBS          = -lcrypto
-SHAREDLIBS    =
+APIVERSION = $(call PKGCFG,apiversion)
 
 ### Allow user defined options to overwrite defaults:
 
--include $(VDRDIR)/Make.global
--include $(VDRDIR)/Make.config
--include Make.config
-
-# we need this ATM because of the helper.h macros...
-CXXFLAGS += -fno-strict-aliasing
+-include $(PLGCFG)
 
-### The version number of VDR (taken from VDR's "config.h"):
+### The name of the distribution archive:
 
-VDRVERSION := $(shell sed -ne '/define VDRVERSION/ s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/include/vdr/config.h)
-APIVERSION := $(shell sed -ne '/define APIVERSION/ s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/include/vdr/config.h)
-ifeq ($(strip $(APIVERSION)),)
-   APIVERSION = $(VDRVERSION)
-endif
-VDRVERSNUM := $(shell sed -ne '/define VDRVERSNUM/ s/^.[a-zA-Z ]*\([0-9]*\) .*$$/\1/p' $(VDRDIR)/include/vdr/config.h)
-APIVERSNUM := $(shell sed -ne '/define APIVERSNUM/ s/^.[a-zA-Z ]*\([0-9]*\) .*$$/\1/p' $(VDRDIR)/include/vdr/config.h)
-ifeq ($(strip $(APIVERSNUM)),)
-   APIVERSNUM = $(VDRVERSNUM)
-endif
+ARCHIVE = $(PLUGIN)-$(VERSION)
+PACKAGE = vdr-$(ARCHIVE)
 
-### The object files (add further files here):
+### The name of the shared object file:
 
-OBJS = $(PLUGIN).o data.o filter.o system.o misc.o cam.o device.o version.o \
-       smartcard.o network.o crypto.o system-common.o parse.o log.o \
-       override.o
+SOFILE = libvdr-$(PLUGIN).so
 
-### Internationalization (I18N):
+### Includes and Defines (add further entries here):
 
-PODIR     = po
-I18Npot   = $(PODIR)/$(PLUGIN).pot
-I18Nmo    = vdr-$(PLUGIN).mo
-I18Nmsgs  = $(addprefix $(LOCALEDIR)/,$(addsuffix /LC_MESSAGES/$(I18Nmo),$(notdir $(foreach file, $(wildcard $(PODIR)/*.po), $(basename $(file))))))
-LOCALEDIR = $(VDRDIR)/locale
+DEFINES  += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
+SHAREDDEFINES = -DAPIVERSNUM=$(APIVERSNUM) -D_GNU_SOURCE
+LIBS          = -lcrypto
 
-### VDR version dependant
+# we need this ATM because of the helper.h macros...
 
-# test VDR version
-BYVERS = $(strip $(shell if test $(APIVERSNUM) -ge 010703; then echo "*"; fi))
-# test if PlayTsVideo() exists (e.g. TSplay patch)
-BYTSPL = $(strip $(shell grep -l 'PlayTsVideo' $(VDRDIR)/include/vdr/device.h))
+CXXFLAGS += -fno-strict-aliasing
 
-ifneq ($(BYVERS)$(BYTSPL),)
-  SHAREDDEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-endif
+### The object files (add further files here):
 
-#
-# generic stuff
-#
+OBJS = $(PLUGIN).o data.o filter.o system.o misc.o cam.o device.o version.o \
+       smartcard.o network.o crypto.o system-common.o parse.o log.o \
+       override.o
 
-# smartcard default port (dropped)
-ifdef DEFAULT_PORT
-  $(error DEFAULT_PORT support was removed, use cardslot.conf)
-endif
+# FFdeCSA
+PARALLEL ?= PARALLEL_128_SSE2
+CSAFLAGS ?= -fexpensive-optimizations -funroll-loops -mmmx -msse -msse2 -msse3
+FFDECSADIR = FFdecsa
+FFDECSA = $(FFDECSADIR)/FFdecsa.o
+DECSALIB = $(FFDECSA)
 
 ifdef WITH_PCSC
   DEFINES  += -DWITH_PCSC
@@ -124,40 +91,25 @@ endif
 HAVE_SD := $(wildcard ../dvbsddevice/dvbsddevice.c)
 ifneq ($(strip $(HAVE_SD)),)
   DEFINES += -DWITH_SDDVB
-  DEVPLUGTARGETS += $(LIBDIR)/libsc-dvbsddevice-$(SCAPIVERS).so.$(APIVERSION)
+  DEVPLUGTARGETS += libdvbapi-dvbsddevice.so
+  DEVPLUGINSTALL += install-devplug-sddvb
 endif
-DEVPLUGOBJS += device-sd.o
 HAVE_HD := $(wildcard ../dvbhddevice/dvbhddevice.c)
 ifneq ($(strip $(HAVE_HD)),)
   HDVERS := $(shell sed -ne '/*VERSION/ s/^.*=.*"\(.*\)".*$$/\1/p' ../dvbhddevice/dvbhddevice.c)
-  ifeq ($(findstring dag,$(HDVERS)),)
+ifeq ($(findstring dag,$(HDVERS)),)
     DEFINES += -DWITH_HDDVB
-    DEVPLUGTARGETS += $(LIBDIR)/libsc-dvbhddevice-$(SCAPIVERS).so.$(APIVERSION)
-  endif
+    DEVPLUGTARGETS += libdvbapi-dvbhddevice.so
+    DEVPLUGINSTALL += install-devplug-hddvb
+endif
 endif
-DEVPLUGOBJS += device-hd.o
 
 # max number of CAIDs per slot
-MAXCAID := $(shell sed -ne '/define MAXCASYSTEMIDS/ s/^.[a-zA-Z ]*\([0-9]*\).*$$/\1/p' $(VDRDIR)/ci.c)
+MAXCAID := 64
 ifneq ($(strip $(MAXCAID)),)
   DEFINES += -DVDR_MAXCAID=$(MAXCAID)
 endif
 
-# FFdeCSA
-CPUOPT     ?= pentium
-PARALLEL   ?= PARALLEL_32_INT
-CSAFLAGS   ?= -Wall -fPIC -g -O3 -mmmx -fomit-frame-pointer -fexpensive-optimizations -funroll-loops
-FFDECSADIR  = FFdecsa
-FFDECSA     = $(FFDECSADIR)/FFdecsa.o
-FFDECSATEST = $(FFDECSADIR)/FFdecsa_test.done
-
-# SASC
-ifdef SASC
-DEFINES += -DSASC
-FFDECSA =
-FFDECSATEST =
-endif
-
 # export for system makefiles
 export SCAPIVERS
 export APIVERSION
@@ -167,22 +119,19 @@ export SHAREDLIBS
 export CXX
 export CXXFLAGS
 
-### Targets:
+### The main target:
 
-ifdef STATIC
-BUILDTARGETS = $(LIBDIR)/libvdr-$(PLUGIN).a
-SHAREDDEFINES += -DSTATICBUILD
-else
-BUILDTARGETS = $(LIBDIR)/libvdr-$(PLUGIN).so.$(APIVERSION) systems-pre $(DEVPLUGTARGETS)
-endif
-BUILDTARGETS += $(FFDECSATEST) systems i18n
+all: $(SOFILE) systems i18n
+.PHONY: i18n systems
 
-all: $(BUILDTARGETS)
-.PHONY: i18n systems systems-pre contrib clean clean-core clean-systems clean-pre dist srcdist
+### Implicit rules:
 
-# Dependencies:
+%.o: %.c
+	$(CXX) $(CXXFLAGS) -c $(DEFINES) $(INCLUDES) -o $@ $<
+
+### Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 DEPFILES = $(subst i18n.c,,$(subst version.c,,$(OBJS:%.o=%.c)))
 $(DEPFILE): $(DEPFILES) $(wildcard *.h)
@@ -190,66 +139,77 @@ $(DEPFILE): $(DEPFILES) $(wildcard *.h)
 
 -include $(DEPFILE)
 
-# Rules
-
-%.o: %.c
-	$(CXX) $(CXXFLAGS) -c $(DEFINES) $(SHAREDDEFINES) $(INCLUDES) $<
+### Internationalization (I18N):
 
-libvdr-$(PLUGIN).so: $(OBJS) $(FFDECSA)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) $(FFDECSA) $(LIBS) $(SHAREDLIBS) -o $@
+PODIR     = po
+I18Npo    = $(wildcard $(PODIR)/*.po)
+I18Nmo    = $(addsuffix .mo, $(foreach file, $(I18Npo), $(basename $(file))))
+I18Nmsgs  = $(addprefix $(DESTDIR)$(LOCDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
+I18Npot   = $(PODIR)/$(PLUGIN).pot
 
-$(LIBDIR)/libvdr-$(PLUGIN).so.$(APIVERSION): libvdr-$(PLUGIN).so
-	@cp -p $< $@
+%.mo: %.po
+	msgfmt -c -o $@ $<
 
-$(LIBDIR)/libvdr-$(PLUGIN).a: $(OBJS)
-	$(AR) r $@ $(OBJS)
+$(I18Npot): $(wildcard *.c)
+	xgettext -C -cTRANSLATORS --no-wrap --no-location -k -ktr -ktrNOOP --package-name=vdr-$(PLUGIN) --package-version=$(VERSION) --msgid-bugs-address='<see README>' -o $@ `ls $^`
 
-libsc-dvbsddevice.so: device-sd.o
-	$(CXX) $(CXXFLAGS) -shared $< $(SHAREDLIBS) -o $@
+%.po: $(I18Npot)
+	msgmerge -U --no-wrap --no-location --backup=none -q -N $@ $<
+	@touch $@
 
-$(LIBDIR)/libsc-dvbsddevice-$(SCAPIVERS).so.$(APIVERSION): libsc-dvbsddevice.so
-	@cp -p $< $@
+$(I18Nmsgs): $(DESTDIR)$(LOCDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
+	install -D -m644 $< $@
 
-libsc-dvbhddevice.so: device-hd.o
-	$(CXX) $(CXXFLAGS) -shared $< $(SHAREDLIBS) -o $@
+.PHONY: i18n
+i18n: $(I18Nmo) $(I18Npot)
 
-$(LIBDIR)/libsc-dvbhddevice-$(SCAPIVERS).so.$(APIVERSION): libsc-dvbhddevice.so
-	@cp -p $< $@
+install-i18n: $(I18Nmsgs)
 
-$(FFDECSA) $(FFDECSATEST): $(FFDECSADIR)/*.c $(FFDECSADIR)/*.h
-	@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CSAFLAGS) -march=$(CPUOPT)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) all
+### Targets:
 
-$(I18Npot): $(shell grep -rl '\(tr\|trNOOP\)(\".*\")' *.c $(SYSDIR))
-	xgettext -C -cTRANSLATORS --no-wrap --no-location -k -ktr -ktrNOOP --package-name=VDR-SC --package-version=$(VERSION) --msgid-bugs-address='<noone@nowhere.org>' -o $@ `ls $^`
+$(SOFILE): $(OBJS) $(FFDECSA)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(OBJS) $(DECSALIB) -o $@
 
-%.po: $(I18Npot)
-	msgmerge -U --no-wrap --no-location --no-fuzzy-matching --backup=none -q $@ $<
-	@touch $@
-
-%.mo: %.po
-	msgfmt -c -o $@ $<
+libsc-dvbsddevice.so: device-sd.o
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $< -o $@
 
-$(I18Nmsgs): $(LOCALEDIR)/%/LC_MESSAGES/$(I18Nmo): $(PODIR)/%.mo
-	@mkdir -p $(dir $@)
-	cp $< $@
+libsc-dvbhddevice.so: device-hd.o
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $< -o $@
 
-i18n: $(I18Nmsgs)
+$(FFDECSA): $(FFDECSADIR)/*.c $(FFDECSADIR)/*.h
+	@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CXXFLAGS) $(LDFLAGS) $(CSAFLAGS)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) all
 
 version.c: FORCE
 	@echo >$@.new "/* generated file, do not edit */"; \
-	 echo >>$@.new 'const char *ScVersion =' '"'$(VERSION)'";'; \
-	 diff $@.new $@ >$@.diff 2>&1; \
-	 if test -s $@.diff; then mv -f $@.new $@; fi; \
-	 rm -f $@.new $@.diff;
+	echo >>$@.new 'const char *ScVersion =' '"'$(VERSION)'";'; \
+	diff $@.new $@ >$@.diff 2>&1; \
+	if test -s $@.diff; then mv -f $@.new $@; fi; \
+	rm -f $@.new $@.diff;
 
 systems:
 	@for i in `ls -A -I ".*" $(SYSDIR)`; do $(MAKE) -f ../../Makefile.system -C "$(SYSDIR)/$$i" all || exit 1; done
 
-systems-pre:
-	@for i in `ls -A -I ".*" $(PREDIR) | grep -- '-$(SCAPIVERS).so.$(APIVERSION)$$'`; do cp -p "$(PREDIR)/$$i" "$(LIBDIR)"; done
+install-lib: $(SOFILE)
+	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
+
+install-devplug-sddvb: libdvbapi-dvbsddevice.so
+	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
+
+install-devplug-hddvb: libdvbapi-dvbhddevice.so
+	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
+
+install-systems:
+	@for i in `ls -A -I ".*" $(SYSDIR)`; do $(MAKE) -f ../../Makefile.system -C "$(SYSDIR)/$$i" install || exit 1; done
+	
+install: install-lib $(DEVPLUGINSTALL) install-systems
 
-contrib:
-	@$(MAKE) -C contrib all
+dist: clean
+	@-rm -rf $(TMPDIR)/$(ARCHIVE)
+	@mkdir $(TMPDIR)/$(ARCHIVE)
+	@cp -a * $(TMPDIR)/$(ARCHIVE)
+	@tar czf $(PACKAGE).tgz -C $(TMPDIR) $(ARCHIVE)
+	@-rm -rf $(TMPDIR)/$(ARCHIVE)
+	@echo Distribution package created as $(PACKAGE).tgz
 
 clean-systems:
 	@for i in `ls -A -I ".*" $(SYSDIR)`; do $(MAKE) -f ../../Makefile.system -C "$(SYSDIR)/$$i" clean; done
@@ -263,33 +223,6 @@ clean-core:
 	@-rm -f $(OBJS) $(DEVPLUGOBJS) $(DEPFILE) version.c *.so *.tar.gz core* *~
 	@-rm -f $(PODIR)/*.mo
 
-clean-pre:
-	@-find "$(PREDIR)" -type f -not -name ".empty" -not -iname "*-$(SCAPIVERS).so.*" | xargs rm -f
-
 clean: clean-core clean-systems
 
-dist: ARCHIVE := $(PLUGIN)-$(RELEASE)
-dist: clean-core
-	@for i in `ls -A -I ".*" $(SYSDIR)`; do $(MAKE) -f ../../Makefile.system -C "$(SYSDIR)/$$i" dist; done
-	@-rm -rf $(TMPDIR)/$(ARCHIVE)
-	@mkdir $(TMPDIR)/$(ARCHIVE)
-	@cp -a * $(TMPDIR)/$(ARCHIVE)
-	@echo -n "release" >$(TMPDIR)/$(ARCHIVE)/$(DISTFILE)
-	@path="$(TMPDIR)/$(ARCHIVE)/$(notdir $(SYSDIR))";\
-	 for i in `ls -A -I ".*" $$path`; do if [ -f "$$path/$$i/nonpublic.mk" ]; then rm -rf "$$path/$$i"; fi; if [ -f "$$path/$$i/nonpublic.sh" ]; then (cd $$path/$$i ; source ./nonpublic.sh ; rm ./nonpublic.sh); fi; done
-	@strip --strip-unneeded --preserve-dates $(TMPDIR)/$(ARCHIVE)/$(notdir $(PREDIR))/* || true
-	@tar czf vdr-$(ARCHIVE).tar.gz -C $(TMPDIR) $(ARCHIVE)
-	@-rm -rf $(TMPDIR)/$(ARCHIVE)
-	@echo Distribution package created as vdr-$(ARCHIVE).tar.gz
-
-copy: ARCHIVE := $(PLUGIN)-$(VERSION)
-copy: clean clean-pre
-	@-rm -rf $(TMPDIR)/$(ARCHIVE)
-	@mkdir $(TMPDIR)/$(ARCHIVE)
-	@cp -a .hgtags .hgignore * $(TMPDIR)/$(ARCHIVE)
-	@echo -n $(SUBREL) | sed -e 's/HG-/CP-/' >$(TMPDIR)/$(ARCHIVE)/$(DISTFILE)
-	@tar czf vdr-$(ARCHIVE).tar.gz -C $(TMPDIR) $(ARCHIVE)
-	@-rm -rf $(TMPDIR)/$(ARCHIVE)
-	@echo Full copy package created as vdr-$(ARCHIVE).tar.gz
-
 FORCE:
diff --git a/Makefile.system b/Makefile.system
index 36a192e..8280115 100644
--- a/Makefile.system
+++ b/Makefile.system
@@ -18,8 +18,8 @@
 
 ### The directory environment:
 
-LIBDIR = ../../../../lib
-PREDIR = ../../systems-pre
+PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
+LIBDIR = $(call PKGCFG,libdir)
 
 ### Includes and Defines
 
@@ -41,24 +41,8 @@ $(DEPFILE): $(OBJS:%.o=%.c) $(wildcard *.h)
 
 LIBSC    = libsc-$(TARGET).so
 LIBSCVER = libsc-$(TARGET)-$(SCAPIVERS).so.$(APIVERSION)
-LIBSCAR  = libsc-$(TARGET).a
 
-ifdef STATIC
-BUILDTARGETS = $(LIBDIR)/$(LIBSCAR)
-else
-BUILDTARGETS = $(LIBDIR)/$(LIBSCVER) $(PREDIR)/$(LIBSCVER)
-endif
-
-define clean
-@-rm -f $(OBJS) $(CLEAN_RM) $(DEPFILE) $(LIBSC) core* *~
-endef
-
-define clean-pre
-@find "$(PREDIR)" -type f -iname "libsc-$(TARGET)-*.so.*" -not -iname "libsc-$(TARGET)-$(SCAPIVERS).so.*" | xargs rm -f
-@-rm -f $(PREDIR)/$(LIBSCVER)
-endef
-
-all: $(BUILDTARGETS)
+all: $(LIBSC)
 .PHONY: clean dist
 
 %.o: %.c
@@ -66,29 +50,9 @@ all: $(BUILDTARGETS)
 
 $(LIBSC): $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) $(LIBS) $(SHAREDLIBS) -o $@
-	$(clean-pre)
-
-$(LIBDIR)/$(LIBSCVER): $(LIBSC)
-	@cp -p $< $@
-
-$(LIBDIR)/$(LIBSCAR): $(OBJS)
-	$(AR) r $@ $(OBJS)
-
-ifdef NONPUBLIC
-$(PREDIR)/$(LIBSCVER): $(LIBSC)
-	$(clean-pre)
-	@cp -p $< $@
-
-dist: $(PREDIR)/$(LIBSCVER)
-	$(clean)
-else
-$(PREDIR)/$(LIBSCVER):
 
-dist:
-	$(clean-pre)
-	$(clean)
-endif
+install: $(LIBSC)
+	install -D $^ $(DESTDIR)$(LIBDIR)/$(LIBSCVER)
 
 clean:
-	$(clean-pre)
-	$(clean)
+	@-rm -f $(OBJS) $(CLEAN_RM) $(DEPFILE) $(LIBSC) core* *~
\ No newline at end of file
