--- cxxtools-2.1.1/src/utf8codec.cpp	2012-05-03 18:06:16.000000000 +0200
+++ b/cxxtools-2.1.1/src/utf8codec.cpp	2013-03-06 11:35:53.994565118 +0100
@@ -307,7 +307,7 @@
         Char* to_next = to;
 
         const char* from_next = from;
-        r = codec.in(state, from, from + size, from_next, to, to + sizeof(to), to_next);
+        r = codec.in(state, from, from + size, from_next, to,  to + sizeof(to) / sizeof(to[0]), to_next);
 
         if (r == error)
             throw ConversionError("character conversion failed");
