diff --git a/mymenurecordings.c b/mymenurecordings.c
index 99e735f..db14feb 100644
--- a/mymenurecordings.c
+++ b/mymenurecordings.c
@@ -160,6 +160,7 @@ eOSState myMenuRecordingInfo::ProcessKey(eKeys Key)
 // --- myMenuRecordingsItem ---------------------------------------------------
 myMenuRecordingsItem::myMenuRecordingsItem(cRecording *Recording,int Level)
 {
+  int savedlevel = Level;
 #if VDRVERSNUM >= 10733
   recording=Recording;
 #endif
@@ -446,6 +447,7 @@ myMenuRecordingsItem::myMenuRecordingsItem(cRecording *Recording,int Level)
         title=strdup("");
     }
   }
+  level = savedlevel;
   SetText(title);
 }
 
