--- plugin.c~	2012-09-12 00:24:40.581589651 +0200
+++ plugin.c	2012-09-12 00:24:40.614921138 +0200
@@ -27,6 +27,7 @@
 #include "extdata.h"
 #include "strfct.h"
 
+#include <vdr/config.h>
 #include <vdr/plugin.h>
 
 #include <ctype.h>
@@ -233,7 +234,11 @@
     }
     while (pos2 != std::string::npos && index < GRAPHLCD_MAX_DISPLAYS);    
 
+#if APIVERSNUM > 10729
+    const char* tempConfigDir = ResourceDirectory(kPluginName);
+#else
     const char* tempConfigDir = ConfigDirectory(kPluginName);
+#endif
     if (!tempConfigDir) {
         return false;
     } else {
