diff -ruN peer-0.0.1.orig/menu.c peer-0.0.1/menu.c
--- peer-0.0.1.orig/menu.c	2015-09-15 23:49:31.591864562 +0200
+++ peer-0.0.1/menu.c	2015-09-15 23:52:06.068770396 +0200
@@ -124,7 +124,7 @@
 		int prio = IDLEPRIORITY;
 		cString name(""), type("");
 		Clear();
-		for (cLine *line = reply->First(); line; line = reply->Next(line)) {
+		for (cLine *line = (cLine*) reply->First(); line; line = (cLine*) reply->Next(line)) {
 			cString key, value;
 			if (Parse(line->Text(), key, value)) {
 				if (strcasecmp("DEV", *key) == 0) {
diff -ruN peer-0.0.1.orig/peer.c peer-0.0.1/peer.c
--- peer-0.0.1.orig/peer.c	2015-09-15 23:49:31.591864562 +0200
+++ peer-0.0.1/peer.c	2015-09-15 23:50:51.400264168 +0200
@@ -160,11 +160,12 @@
 				buffer.Append("PLAY = %s\n", replay);
 			}
 			else if (!replay && (d->IsPrimaryDevice() || d == cDevice::ActualDevice())) {
-				cChannel *c = Channels.GetByNumber(cDevice::CurrentChannel());
+                                LOCK_CHANNELS_READ;
+				cChannel *c = (cChannel*) Channels->GetByNumber(cDevice::CurrentChannel());
 				if (c)
 					buffer.Append("LIVE = %s\n", c->Name());
 			}
-			for (cPeerStatusRecording * r = recs->First(); r; r = recs->Next(r)) {
+			for (const cPeerStatusRecording * r = recs->First(); r; r = recs->Next(r)) {
 				if (d == r->Device())
 					buffer.Append("REC = %s\n", r->Name());
 
