diff -ruN dddvb/ddbridge/ddbridge-core.c dddvb.edit/ddbridge/ddbridge-core.c
--- dddvb/ddbridge/ddbridge-core.c	2014-04-25 15:31:54.485174885 +0200
+++ dddvb.edit/ddbridge/ddbridge-core.c	2014-04-25 15:36:14.096574714 +0200
@@ -47,7 +47,6 @@
 
 #include "ddbridge-mod.c"
 #include "ddbridge-i2c.c"
-#include "ddbridge-ns.c"
 
 
 static void ddb_set_dma_table(struct ddb *dev, struct ddb_dma *dma)
@@ -878,12 +877,6 @@
 /****************************************************************************/
 /****************************************************************************/
 
-static long mod_ioctl(struct file *file,
-		      unsigned int cmd, unsigned long arg)
-{
-	return dvb_usercopy(file, cmd, arg, ddbridge_mod_do_ioctl);
-}
-
 static const struct file_operations mod_fops = {
 	.owner   = THIS_MODULE,
 	.read    = ts_read,
@@ -892,7 +885,6 @@
 	.release = mod_release,
 	.poll    = ts_poll,
 	.mmap    = 0,
-	.unlocked_ioctl = mod_ioctl,
 };
 
 static struct dvb_device dvbdev_mod = {
@@ -1385,10 +1377,6 @@
 		dvb_frontend_detach(dvb->fe);
 		dvb->fe = dvb->fe2 = NULL;
 		/* fallthrough */
-	case 0x21:
-		if (input->port->dev->ns_num)
-			dvb_netstream_release(&dvb->dvbns);
-		/* fallthrough */
 	case 0x20:
 		dvb_net_release(&dvb->dvbnet);
 		/* fallthrough */
@@ -1538,12 +1526,6 @@
 		return ret;
 	dvb->attached = 0x20;
 
-	if (input->port->dev->ns_num) {
-		ret = netstream_init(input);
-		if (ret < 0)
-			return ret;
-		dvb->attached = 0x21;
-	}
 	dvb->fe = dvb->fe2 = 0;
 	switch (port->type) {
 	case DDB_TUNER_DVBS_ST:
diff -ruN dddvb/ddbridge/ddbridge.h dddvb.edit/ddbridge/ddbridge.h
--- dddvb/ddbridge/ddbridge.h	2014-04-25 15:31:54.485174885 +0200
+++ dddvb.edit/ddbridge/ddbridge.h	2014-04-25 15:32:11.877485538 +0200
@@ -69,7 +69,6 @@
 #include <linux/device.h>
 #include <linux/io.h>
 
-#include "dvb_netstream.h"
 #include "dmxdev.h"
 #include "dvbdev.h"
 #include "dvb_demux.h"
@@ -198,7 +197,6 @@
 	struct dmxdev          dmxdev;
 	struct dvb_demux       demux;
 	struct dvb_net         dvbnet;
-	struct dvb_netstream   dvbns;
 	struct dmx_frontend    hw_frontend;
 	struct dmx_frontend    mem_frontend;
 	int                    users;
diff -ruN dddvb/ddbridge/Kbuild dddvb.edit/ddbridge/Kbuild
--- dddvb/ddbridge/Kbuild	2014-04-25 15:31:54.485174885 +0200
+++ dddvb.edit/ddbridge/Kbuild	2014-04-25 15:32:11.877485538 +0200
@@ -4,5 +4,5 @@
 obj-$(CONFIG_DVB_OCTONET) += octonet.o 
 
 EXTRA_CFLAGS += -Idrivers/media/dvb/frontends -Idrivers/media/dvb-frontends
-EXTRA_CFLAGS += -Idrivers/media/common/tuners 
+EXTRA_CFLAGS += -Idrivers/media/common/tuners -Idrivers/media/dvb-core -Idrivers/staging/media/cxd2099 -Iinclude/uapi/linux/dvb -Idrivers/media/tuners
 NOSTDINC_FLAGS += -I$(SUBDIRS)/frontends -I$(SUBDIRS)/include -I$(SUBDIRS)/dvb-core 
\ No newline at end of file
diff -ruN dddvb/frontends/cxd2843.c dddvb.edit/frontends/cxd2843.c
--- dddvb/frontends/cxd2843.c	2014-03-09 11:22:26.653432931 +0100
+++ dddvb.edit/frontends/cxd2843.c	2014-03-09 11:23:18.513433935 +0100
@@ -1078,9 +1078,6 @@
 	case SYS_DVBC_ANNEX_A:
 		state->omode = OM_DVBC;
 		break;
-	case SYS_DVBC2:
-		state->omode = OM_DVBC2;
-		break;
 	case SYS_DVBT:
 		state->omode = OM_DVBT;
 		break;
@@ -1556,7 +1553,7 @@
 }
 
 static struct dvb_frontend_ops common_ops_2843 = {
-	.delsys = { SYS_DVBC_ANNEX_A, SYS_DVBT, SYS_DVBT2, SYS_DVBC2 },
+	.delsys = { SYS_DVBC_ANNEX_A, SYS_DVBT, SYS_DVBT2 },
 	.info = {
 		.name = "CXD2843 DVB-C/C2 DVB-T/T2",
 		.frequency_stepsize = 166667,	/* DVB-T only */
diff -ruN dddvb/frontends/tda18212dd.c dddvb.edit/frontends/tda18212dd.c
--- dddvb/frontends/tda18212dd.c	2014-03-09 11:22:26.653432931 +0100
+++ dddvb.edit/frontends/tda18212dd.c	2014-03-09 11:23:31.510100853 +0100
@@ -777,8 +777,7 @@
 	/*pr_info("tuner bw=%u  freq=%u\n", bw, state->m_Frequency);*/
 	if (p->delivery_system == SYS_DVBT ||
 	    p->delivery_system == SYS_DVBT2 ||
-	    p->delivery_system == SYS_ISDBT ||
-	    p->delivery_system == SYS_DVBC2) {
+	    p->delivery_system == SYS_ISDBT) {
 		switch (bw) {
 		case 6:
 			Standard = HF_DVBT_6MHZ;
