diff -ruN skinelchi-0.2.8/bitmap.c skinelchi-0.2.8.edit/bitmap.c
--- skinelchi-0.2.8/bitmap.c	2011-09-18 14:26:54.000000000 +0200
+++ skinelchi-0.2.8.edit/bitmap.c	2013-08-13 16:50:40.000000000 +0200
@@ -138,9 +138,9 @@
       bmp.SetBpp((colors <= 16) ? 4 : 8);
       const PixelPacket *pixels = imgkConv.getConstPixels(0, 0, w, h);
       col = (0xFF << 24)
-            | ( (pixels->green * 255 / MaxRGB) << 8)
-            | ( (pixels->red * 255 / MaxRGB) << 16)
-            | ( (pixels->blue * 255 / MaxRGB) );
+            | (int (pixels->green * 255 / MaxRGB) << 8)
+            | (int (pixels->red * 255 / MaxRGB) << 16)
+            | (int (pixels->blue * 255 / MaxRGB) );
       for (int iy = 0; iy < yoff; iy++)
          for (int ix = 0; ix < x; ix++)
             bmp.DrawPixel(ix, iy, col);
@@ -156,9 +156,9 @@
       for (int iy = yoff; iy < (h + yoff); iy++) {
          for (int ix = xoff; ix < (w + xoff); ix++) {
             col = (0xFF << 24)
-               | ( (pixels->green * 255 / MaxRGB) << 8)
-               | ( (pixels->red * 255 / MaxRGB) << 16)
-               | ( (pixels->blue * 255 / MaxRGB) );
+               | (int (pixels->green * 255 / MaxRGB) << 8)
+               | (int (pixels->red * 255 / MaxRGB) << 16)
+               | (int (pixels->blue * 255 / MaxRGB) );
             bmp.DrawPixel(ix, iy, col);
             pixels++;
          }
@@ -172,9 +172,9 @@
       for (int iy = 0; iy < h; iy++) {
          for (int ix = 0; ix < w; ix++) {
             col = (0xFF << 24)
-               | ( (pixels->green * 255 / MaxRGB) << 8)
-               | ( (pixels->red * 255 / MaxRGB) << 16)
-               | ( (pixels->blue * 255 / MaxRGB) );
+               | (int (pixels->green * 255 / MaxRGB) << 8)
+               | (int (pixels->red * 255 / MaxRGB) << 16)
+               | (int (pixels->blue * 255 / MaxRGB) );
             bmp.DrawPixel(ix, iy, col);
             pixels++;
          }
diff -ruN skinelchi-0.2.8/Makefile skinelchi-0.2.8.edit/Makefile
--- skinelchi-0.2.8/Makefile	2013-08-13 16:51:03.959306398 +0200
+++ skinelchi-0.2.8.edit/Makefile	2013-08-13 16:52:11.645276882 +0200
@@ -110,5 +110,5 @@
 	DEFINES += -DSKINELCHI_HAVE_IMAGEMAGICK
-	INCLUDES += $(shell pkg-config --cflags-only-I ImageMagick++)
+	INCLUDES += $(shell pkg-config --cflags ImageMagick++)
 endif
 
 ifdef SKINELCHI_DEBUG
 
