diff -ruN skinelchi-0.2.8/skinelchi.c skinelchi-0.2.8.edited/skinelchi.c
--- skinelchi-0.2.8/skinelchi.c	2013-02-10 19:27:17.000000000 +0100
+++ skinelchi-0.2.8.edited/skinelchi.c	2013-08-16 08:53:32.171314518 +0200
@@ -159,7 +159,7 @@
 
    // set default logo dir if not supplied by commandline arguments
    if (isempty(ElchiConfig.GetLogoBaseDir()))
-      ElchiConfig.SetLogoBaseDir(cPlugin::ConfigDirectory(PLUGIN_NAME_I18N));
+      ElchiConfig.SetLogoBaseDir(cPlugin::ResourceDirectory(PLUGIN_NAME_I18N));
 
    isyslog("SkinElchi: using channel logo base dir '%s'", ElchiConfig.GetLogoBaseDir());
 #ifdef SKINELCHI_HAVE_IMAGEMAGICK
@@ -221,7 +221,7 @@
           "                                       (default: '/video/epgimages')\n"
 #endif
           "  -l PathToLogos, --logos=PathToLogos  optional path for XPM channel logos\n"
-          "                                       (default: '<configDir>/skinelchi/logos')\n";
+          "                                       (default: '<resourceDir>/skinelchi/logos')\n";
 }
 
 bool cPluginSkinElchi::ProcessArgs(int argc, char *argv[])
