diff --git a/codec.c b/codec.c
index be6ed29..525baca 100644
--- a/codec.c
+++ b/codec.c
@@ -226,7 +226,7 @@ static int Codec_get_buffer(AVCodecContext * video_ctx, AVFrame * frame)
 	if (video_ctx->pkt) {
 	    frame->pkt_pts = video_ctx->pkt->pts;
 	} else {
-	    frame->pkt_pts = AV_NOPTS_VALUE;
+	    //frame->pkt_pts = AV_NOPTS_VALUE;
 	}
 	return 0;
     }
@@ -253,7 +253,7 @@ static int Codec_get_buffer(AVCodecContext * video_ctx, AVFrame * frame)
 	if (video_ctx->pkt) {
 	    frame->pkt_pts = video_ctx->pkt->pts;
 	} else {
-	    frame->pkt_pts = AV_NOPTS_VALUE;
+	    //frame->pkt_pts = AV_NOPTS_VALUE;
 	}
 	return 0;
     }
