diff --git a/displaychannel.c b/displaychannel.c
index 73ccc73..1c8132f 100644
--- a/displaychannel.c
+++ b/displaychannel.c
@@ -11,8 +11,8 @@ cNopacityDisplayChannel::cNopacityDisplayChannel(cImageCache *imgCache, bool Wit
     if (firstDisplay) {
         imgCache->CreateCacheDelayed();
         firstDisplay = false;
-        doOutput = false;
-        return;
+        doOutput = true;
+        //return;
     } else
         doOutput = true;
     groupSep = false;
