diff -ruN vdr-plugin-boblight-0.0.3/main_menu.c vdr-plugin-boblight-0.0.3.edit/main_menu.c
--- vdr-plugin-boblight-0.0.3/main_menu.c	2013-12-29 17:58:48.000000000 +0100
+++ vdr-plugin-boblight-0.0.3.edit/main_menu.c	2014-01-02 19:19:39.358027876 +0100
@@ -14,6 +14,7 @@
 }
 
 void cBoblightPluginMenu::Create(void) {
+   SetMenuCategory(mcPluginSetup);
    Clear();
 
    Add(new cMenuEditStraItem(tr("View Mode"), &cfg.viewMode, cAmbiService::vmCount, cAmbiService::viewModes));
