diff -ruN tvguide-1.0.0/tvguide.c tvguide-1.0.0.edited/tvguide.c
--- tvguide-1.0.0/tvguide.c	2013-07-23 20:06:31.000000000 +0200
+++ tvguide-1.0.0.edited/tvguide.c	2013-08-16 09:10:22.504793183 +0200
@@ -124,7 +124,7 @@
 bool cPluginTvguide::Start(void)
 {
     if (!logoPathSet) {
-        cString path = cString::sprintf("%s/channellogos/", cPlugin::ConfigDirectory(PLUGIN_NAME_I18N));
+        cString path = cString::sprintf("%s/channellogos/", cPlugin::ResourceDirectory(PLUGIN_NAME_I18N));
         tvguideConfig.SetLogoPath(path);
         logoPathSet = true;
     }
