diff --git a/pages/recordings.ecpp b/pages/recordings.ecpp
index 80f145b..6da5285 100644
--- a/pages/recordings.ecpp
+++ b/pages/recordings.ecpp
@@ -79,7 +79,11 @@ for (deletions_type::const_iterator it = deletions.begin(); it != deletions.end(
 deletions.clear();
 
 int FreeMB, UsedMB;
+#if APIVERSNUM > 20101
+int Percent = cVideoDirectory::VideoDiskSpace(&FreeMB, &UsedMB);
+#else
 int Percent = VideoDiskSpace(&FreeMB, &UsedMB);
+#endif
 int Minutes = int(double(FreeMB) / MB_PER_MINUTE);
 int Hours = Minutes / 60;
 Minutes %= 60;
diff --git a/recman.cpp b/recman.cpp
index 6a2fc00..9ed88f0 100644
--- a/recman.cpp
+++ b/recman.cpp
@@ -112,7 +112,11 @@ namespace vdrlive {
 		if (found == string::npos)
 			return false;
 
+#if APIVERSNUM > 20101
+		string newname = string(cVideoDirectory::Name()) + "/" + name + oldname.substr(found);
+#else
 		string newname = string(VideoDirectory) + "/" + name + oldname.substr(found);
+#endif
 
 		if (!MoveDirectory(oldname.c_str(), newname.c_str(), copy)) {
 			esyslog("[LIVE]: renaming failed from '%s' to '%s'", oldname.c_str(), newname.c_str());
diff --git a/tools.cpp b/tools.cpp
index c744d06..cb3529e 100644
--- a/tools.cpp
+++ b/tools.cpp
@@ -365,7 +365,11 @@ namespace vdrlive {
 			stat(source.c_str(), &st1);
 			stat(target.c_str(),&st2);
 			if (!copy && (st1.st_dev == st2.st_dev)) {
+#if APIVERSNUM > 20101
+				if (!cVideoDirectory::RenameVideoFile(source.c_str(), target.c_str())) {
+#else
 				if (!RenameVideoFile(source.c_str(), target.c_str())) {
+#endif
 					esyslog("[LIVE]: rename failed from %s to %s", source.c_str(), target.c_str());
 					return false;
 				}
@@ -461,7 +465,11 @@ namespace vdrlive {
 						size_t found = source.find_last_of(delim);
 						if (found != std::string::npos) {
 							source = source.substr(0, found);
+#if APIVERSNUM > 20101
+							while (source != cVideoDirectory::Name()) {
+#else
 							while (source != VideoDirectory) {
+#endif
 								found = source.find_last_of(delim);
 								if (found == std::string::npos)
 									break;
@@ -478,7 +486,11 @@ namespace vdrlive {
 					size_t found = target.find_last_of(delim);
 					if (found != std::string::npos) {
 						target = target.substr(0, found);
+#if APIVERSNUM > 20101
+						while (target != cVideoDirectory::Name()) {
+#else
 						while (target != VideoDirectory) {
+#endif
 							found = target.find_last_of(delim);
 							if (found == std::string::npos)
 								break;
