diff --git a/config.c b/config.c
index f7384e8..9d5c1da 100644
--- a/config.c
+++ b/config.c
@@ -703,9 +703,15 @@ config_t::config_t() {
   ff_h264_skip_loop_filter     = FF_H264_SKIP_LOOPFILTER_auto;
 
   strn0cpy(media_root_dir,    "/",            sizeof(media_root_dir));
+#if APIVERSNUM > 20101
+  strn0cpy(browse_files_dir,  cVideoDirectory::Name(), sizeof(browse_files_dir));
+  strn0cpy(browse_music_dir,  cVideoDirectory::Name(), sizeof(browse_music_dir));
+  strn0cpy(browse_images_dir, cVideoDirectory::Name(), sizeof(browse_images_dir));
+#else
   strn0cpy(browse_files_dir,  VideoDirectory, sizeof(browse_files_dir));
   strn0cpy(browse_music_dir,  VideoDirectory, sizeof(browse_music_dir));
   strn0cpy(browse_images_dir, VideoDirectory, sizeof(browse_images_dir));
+#endif
   show_hidden_files = 0;
   cache_implicit_playlists = 1;
   enable_id3_scanner = 1;
diff --git a/menu.c b/menu.c
index c0a152f..3de34a7 100644
--- a/menu.c
+++ b/menu.c
@@ -220,7 +220,11 @@ void cMenuBrowseFiles::Set(void)
   }
 
   if (m_CurrentDir[0] != '/') {
+#if APIVERSNUM > 20101
+    m_CurrentDir = cVideoDirectory::Name();
+#else
     m_CurrentDir = VideoDirectory;
+#endif
   }
 
   // find deepest accessible directory from path
diff --git a/tools/udp_pes_scheduler.c b/tools/udp_pes_scheduler.c
index 615b43b..956aca1 100644
--- a/tools/udp_pes_scheduler.c
+++ b/tools/udp_pes_scheduler.c
@@ -635,7 +635,11 @@ void cUdpScheduler::Send_SAP(bool Announce)
 #if 1
   /* store copy of SDP data */
   if(m_fd_sap < 0) {
+#if APIVERSNUM > 20101
+    cString fname = AddDirectory(cVideoDirectory::Name(),
+#else
     cString fname = AddDirectory(VideoDirectory,
+#endif
 				 cString::sprintf("xineliboutput@%s.sdp",
 						  ip));
     FILE *fp = fopen(fname, "w");
