diff -ruN tvscraper-0.2.0/config.c tvscraper-0.2.0.patched/config.c
--- tvscraper-0.2.0/config.c	2013-08-23 15:22:57.000000000 +0200
+++ tvscraper-0.2.0.patched/config.c	2015-09-16 18:12:48.491095989 +0200
@@ -18,7 +18,8 @@
 }
 
 bool cTVScraperConfig::ChannelActive(int channelNum) {
-    cChannel *channel = Channels.GetByNumber(channelNum);
+    LOCK_CHANNELS_READ;
+    cChannel *channel = (cChannel*) Channels->GetByNumber(channelNum);
     if (channel) {
         string channelID = "";
         channelID = *(channel->GetChannelID().ToString());
diff -ruN tvscraper-0.2.0/setup.c tvscraper-0.2.0.patched/setup.c
--- tvscraper-0.2.0/setup.c	2013-08-24 09:03:52.000000000 +0200
+++ tvscraper-0.2.0.patched/setup.c	2015-09-16 18:15:45.823905250 +0200
@@ -6,7 +6,8 @@
 
 cTVScraperSetup::cTVScraperSetup(cTVScraperWorker *workerThread) {
     worker = workerThread;
-    int numChannels = Channels.Count();
+    LOCK_CHANNELS_READ;
+    int numChannels = Channels->Count();
     for (int i=0; i<numChannels; i++) {
         int akt = 0;
         if (config.ChannelActive(i+1))
@@ -60,7 +61,8 @@
     int numChannels = channelsScrap.size();
     for (int i=0; i<numChannels; i++) {
         if (channelsScrap[i] == 1) {
-            cChannel *channel = Channels.GetByNumber(i+1);
+            LOCK_CHANNELS_READ;
+            cChannel *channel = (cChannel*) Channels->GetByNumber(i+1);
             if (channel) {
                 string channelID = *(channel->GetChannelID().ToString());
                 channelsToScrap << channelID << ";";
@@ -89,7 +91,8 @@
     int currentItem = Current();
     Clear();
     int i=0;
-    for (cChannel *channel = Channels.First(); channel; channel = Channels.Next(channel)) {
+    LOCK_CHANNELS_READ;
+    for (const cChannel *channel = Channels->First(); channel; channel = Channels->Next(channel)) {
         if (!channel->GroupSep()) {
             Add(new cMenuEditBoolItem(channel->Name(), &channelsScrap->at(i), tr("don't scrap"), tr("scrap")));
             i++;
diff -ruN tvscraper-0.2.0/worker.c tvscraper-0.2.0.patched/worker.c
--- tvscraper-0.2.0/worker.c	2013-10-06 22:36:32.000000000 +0200
+++ tvscraper-0.2.0.patched/worker.c	2015-09-16 18:14:43.006200224 +0200
@@ -130,13 +130,14 @@
     int numChannels = channels.size();
     for (int i=0; i<numChannels; i++) {
         string channelID = channels[i];
-        const cChannel *channel = Channels.GetByChannelID(tChannelID::FromString(channelID.c_str()));
+        LOCK_CHANNELS_READ;
+        const cChannel *channel = Channels->GetByChannelID(tChannelID::FromString(channelID.c_str()));
         if (!channel)
             continue;
         dsyslog("tvscraper: scraping Channel %s %s", channel->Name(), channelID.c_str());
         cSchedulesLock schedulesLock;
-        const cSchedules *schedules = cSchedules::Schedules(schedulesLock);
-        const cSchedule *Schedule = schedules->GetSchedule(channel);
+        LOCK_SCHEDULES_READ;
+        const cSchedule *Schedule = Schedules->GetSchedule(channel);
         if (Schedule) {
             const cEvent *event = NULL;
             for (event = Schedule->Events()->First(); event; event =  Schedule->Events()->Next(event)) {
@@ -160,7 +161,8 @@
 
 void cTVScraperWorker::ScrapRecordings(void) {
     db->ClearRecordings();
-    for (cRecording *rec = Recordings.First(); rec; rec = Recordings.Next(rec)) {
+    LOCK_RECORDINGS_READ;
+    for (const cRecording *rec = Recordings->First(); rec; rec = Recordings->Next(rec)) {
         if (overrides->IgnorePath(rec->FileName()))
             continue;
         const cRecordingInfo *recInfo = rec->Info();
@@ -188,7 +190,8 @@
 }
 
 void cTVScraperWorker::CheckRunningTimers(void) {
-    for (cTimer *timer = Timers.First(); timer; timer = Timers.Next(timer)) {
+    LOCK_TIMERS_READ;
+    for (const cTimer *timer = Timers->First(); timer; timer = Timers->Next(timer)) {
         if (timer->Recording()) {
             const cEvent *event = timer->Event();
             if (!event)
