diff -ruN vdr-plugin-chanman-0.0.10.orig/chanman.cc vdr-plugin-chanman-0.0.10/chanman.cc
--- vdr-plugin-chanman-0.0.10.orig/chanman.cc	2015-09-14 20:12:31.097156213 +0200
+++ vdr-plugin-chanman-0.0.10/chanman.cc	2015-09-15 21:21:51.229792822 +0200
@@ -55,9 +55,10 @@
 
 void cMyChannels::Set() {
 	int pos = 1;
-	for (cChannel *channel = Channels.First(); channel;
-			channel = Channels.Next(channel)) {
-		Add(new cMyChannel(channel, pos));
+        LOCK_CHANNELS_READ;
+	for (const cChannel *channel = Channels->First(); channel;
+			channel = Channels->Next(channel)) {
+		Add(new cMyChannel( (cChannel*)channel, pos));
 		pos++;
 	}
 }
@@ -329,8 +330,9 @@
 	bool isInGroup = false;
 
 	cMyChannel *first = NULL;
-	cChannel *currentChannel;
-	currentChannel = Channels.GetByNumber(cDevice::CurrentChannel());
+	const cChannel *currentChannel;
+        LOCK_CHANNELS_READ;
+	currentChannel = Channels->GetByNumber(cDevice::CurrentChannel());
 
 	cItemChoice1 *currentItem = NULL;
 
@@ -703,8 +705,9 @@
 	cMyChannel *lchannel;
 	number = 0;
 	lchannel = firstlch;
-	cChannel *currentChannel;
-	currentChannel = Channels.GetByNumber(cDevice::CurrentChannel());
+	const cChannel *currentChannel;
+        LOCK_CHANNELS_READ;
+	currentChannel = Channels->GetByNumber(cDevice::CurrentChannel());
 	cItemChoice2 *currentItem = NULL;
 
 	for (int i = 1; i <= numch; i++) {
@@ -746,7 +749,8 @@
 
 void cSecondMenu::Propagate(void) {
 	int i = 1;
-	Channels.ReNumber();
+        LOCK_CHANNELS_WRITE;
+	Channels->ReNumber();
 	for (cItemChoice2 *ci = (cItemChoice2 *) First(); ci; ci =
 			(cItemChoice2 *) ci->Next()) {
 		ci->SetPos(i);
@@ -754,13 +758,14 @@
 		ci->Set();
 	}
 	Display();
-	Channels.SetModified(true);
+        Channels->SetModifiedByUser();
 	MyChannels.SetModified();
 }
 
 void cSecondMenu::Move(int From, int To) {
 	int CurrentChannelNr = cDevice::CurrentChannel();
-	cChannel *CurrentChannel = Channels.GetByNumber(CurrentChannelNr);
+        LOCK_CHANNELS_WRITE;
+	const cChannel *CurrentChannel = Channels->GetByNumber(CurrentChannelNr);
 	cChannel *FromChannel = GetChannel(From); // From e' la posizione nel menu di cItemChoice2
 	cChannel *ToChannel = GetChannel(To); // GetChannel Restituisce la posizione nella lista Channels.
 	cMyChannel *FromMyChannel = GetMyChannel(From);
@@ -769,7 +774,7 @@
 	if (FromChannel && ToChannel) {
 		int FromNumber = FromChannel->Number();
 		int ToNumber = ToChannel->Number();
-		Channels.Move(FromChannel, ToChannel);
+		Channels->Move(FromChannel, ToChannel);
 		MyChannels.Move(FromMyChannel, ToMyChannel);
 		cOsdMenu::Move(From, To);
 		Propagate();
@@ -778,7 +783,7 @@
 		if (CurrentChannel && CurrentChannel->Number() != CurrentChannelNr) {
 			if (!cDevice::PrimaryDevice()->Replaying()
 					|| cDevice::PrimaryDevice()->Transferring())
-				Channels.SwitchTo(CurrentChannel->Number());
+				Channels->SwitchTo(CurrentChannel->Number());
 			else
 				cDevice::SetCurrentChannel(CurrentChannel);
 		}
@@ -808,7 +813,8 @@
 
 		FromChannel = FromMyChannel->GetCh();
 
-		Channels.Move(FromChannel, ToChannel);
+                LOCK_CHANNELS_WRITE;
+		Channels->Move(FromChannel, ToChannel);
 		MyChannels.Move(FromMyChannel, ToMyChannel);
 
 		//void cOsdMenu::Add(cOsdItem *Item, bool Current, cOsdItem *After)
