diff -ruN v4l-dvb-saa716x-20120304/linux/drivers/media/common/saa716x/saa716x_ff_cmd.h v4l-dvb-saa716x-20120304.patched/linux/drivers/media/common/saa716x/saa716x_ff_cmd.h
--- v4l-dvb-saa716x-20120304/linux/drivers/media/common/saa716x/saa716x_ff_cmd.h	2012-03-04 18:01:07.000000000 +0100
+++ v4l-dvb-saa716x-20120304.patched/linux/drivers/media/common/saa716x/saa716x_ff_cmd.h	2012-06-05 20:01:17.278360376 +0200
@@ -1,6 +1,24 @@
 #ifndef __SAA716x_FF_CMD_H
 #define __SAA716x_FF_CMD_H
 
+#if !defined OSD_RAW_CMD
+typedef struct osd_raw_cmd_s {
+    const void *cmd_data;
+    int cmd_len;
+    void *result_data;
+    int result_len;
+} osd_raw_cmd_t;
+
+typedef struct osd_raw_data_s {
+    const void *data_buffer;
+    int data_length;
+    int data_handle;
+} osd_raw_data_t;
+
+#define OSD_RAW_CMD            _IOWR('o', 162, osd_raw_cmd_t)
+#define OSD_RAW_DATA           _IOWR('o', 163, osd_raw_data_t)
+#endif
+
 extern int sti7109_cmd_init(struct sti7109_dev *sti7109);
 extern int sti7109_raw_cmd(struct sti7109_dev * sti7109,
 			   osd_raw_cmd_t * cmd);
diff -ruN v4l-dvb-saa716x-20120304/linux/drivers/media/common/saa716x/saa716x_ff_ir.c v4l-dvb-saa716x-20120304.patched/linux/drivers/media/common/saa716x/saa716x_ff_ir.c
--- v4l-dvb-saa716x-20120304/linux/drivers/media/common/saa716x/saa716x_ff_ir.c	2012-03-04 18:01:07.000000000 +0100
+++ v4l-dvb-saa716x-20120304.patched/linux/drivers/media/common/saa716x/saa716x_ff_ir.c	2012-06-05 19:59:20.158355127 +0200
@@ -20,6 +20,7 @@
  *
  */
 
+#include <linux/version.h>
 #include <linux/types.h>
 #include <linux/input.h>
 
diff -ruN v4l-dvb-saa716x-20120304/linux/drivers/media/common/saa716x/saa716x_ff_main.c v4l-dvb-saa716x-20120304.patched/linux/drivers/media/common/saa716x/saa716x_ff_main.c
--- v4l-dvb-saa716x-20120304/linux/drivers/media/common/saa716x/saa716x_ff_main.c	2012-03-04 18:01:07.000000000 +0100
+++ v4l-dvb-saa716x-20120304.patched/linux/drivers/media/common/saa716x/saa716x_ff_main.c	2012-06-05 19:59:25.271688687 +0200
@@ -1,3 +1,4 @@
+#include <linux/version.h>
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/kernel.h>
