--- runvdr	2009-06-28 18:52:37.000000000 +0200
+++ runvdr.patched	2012-06-10 16:26:27.342712317 +0200
@@ -629,10 +629,10 @@
     GetChilds "$1" "$2"
 
     echo -n "Sending ${#childlist[*]} processes the TERM signal."
-    $KILL -TERM ${childlist[*]} >&- 2>&-
+    $KILL -TERM ${childlist[*]} 2>&1> /dev/null
 
     for ((i=0;i<$3;i++)) ; do
-        $PS ${childlist[*]} >&- 2>&- || { echo terminated. ; return ; }
+        $PS ${childlist[*]} 2>&1> /dev/null || { echo terminated. ; return ; }
 
         echo -n .
         $SLEEP 1
@@ -640,10 +640,10 @@
     echo
 
     echo -n "Sending ${#childlist[*]} processes the KILL signal."
-    $KILL -KILL ${childlist[*]} >&- 2>&-
+    $KILL -KILL ${childlist[*]} 2>&1> /dev/null
 
     for ((i=0;i<$4;i++)) ; do
-        $PS ${childlist[*]} >&- 2>&- || { echo terminated. ; return ; }
+        $PS ${childlist[*]} 2>&1> /dev/null || { echo terminated. ; return ; }
         echo -n .
         $SLEEP 1
     done
@@ -746,7 +746,7 @@
         echo -n "Terminating runvdr (PID=$TERMINATE)"
         $KILL -TERM $TERMINATE
         while [ -z "$WAIT" ] || [ "$WAIT" -gt 0 ] ; do
-            $PS $TERMINATE >&- || { echo done ; break ; }
+            $PS $TERMINATE 2>&1> /dev/null || { echo done ; break ; }
             echo -n "."
             $SLEEP 1
             [ -n "$WAIT" ] && let WAIT=WAIT-1
@@ -1003,9 +1003,9 @@
 done
 
 # Clean up PID file
-[ -n "$RUNVDRPID" ] && rm $RUNVDRPID >&- 2>&-
+[ -n "$RUNVDRPID" ] && rm $RUNVDRPID 2>&1> /dev/null
 
 # Clean up proxy file, if used
-[ -n "$PROXYFILE" ] && rm $PROXYFILE >&- 2>&-
+[ -n "$PROXYFILE" ] && rm $PROXYFILE 2>&1> /dev/null
 
 
