diff -ruN cecremote-1.3.0.orig/cecstatusmonitor.cc cecremote-1.3.0/cecstatusmonitor.cc
--- cecremote-1.3.0.orig/cecstatusmonitor.cc	2015-09-14 20:03:06.687290068 +0200
+++ cecremote-1.3.0/cecstatusmonitor.cc	2015-09-14 20:03:12.827237724 +0200
@@ -29,7 +29,8 @@
     }
     if (Device->IsPrimaryDevice()) {
         Dsyslog("Primary device, Channel Switch %d %c", ChannelNumber,l);
-        const cChannel* channel = Channels.GetByNumber(ChannelNumber);
+        LOCK_CHANNELS_READ;
+        const cChannel* channel = Channels->GetByNumber(ChannelNumber);
         if (channel != NULL) {
             if (channel->Vpid() == 0) {
                 Dsyslog("  Radio : %s", channel->Name());
