diff --git a/config.sh b/config.sh
old mode 100644
new mode 100755
diff --git a/reader-common.c b/reader-common.c
index 34de185..4358b8e 100644
--- a/reader-common.c
+++ b/reader-common.c
@@ -246,7 +246,7 @@ bool cardreader_init(struct s_reader *reader) {
 	while (cardreader_device_init(reader) == 2) {
 		int8_t i = 0;
 		do {
-			cs_sleepms(2000);
+			cs_sleepms(20);
 			if (!ll_contains(configured_readers, reader) || !is_valid_client(client) || reader->enable != 1)
 				return false;
 			i++;
diff --git a/webif/pages_mkdep b/webif/pages_mkdep
old mode 100644
new mode 100755
