Index: vdr-plugin-externalplayer-0.1.0.99/Makefile
===================================================================
--- vdr-plugin-externalplayer-0.1.0.99.orig/Makefile	2012-03-27 15:28:50.143843314 +0200
+++ vdr-plugin-externalplayer-0.1.0.99/Makefile	2012-03-27 15:29:07.903843324 +0200
@@ -46,7 +46,7 @@
 
 ### The object files (add further files here):
 
-OBJS = externalplayer.o externalplayer-config.o externalplayer-control.o  externalplayer-i18n.o externalplayer-player.o 
+OBJS = externalplayer.o externalplayer-config.o externalplayer-control.o externalplayer-player.o 
 
 ### Implicit rules:
 
Index: vdr-plugin-externalplayer-0.1.0.99/externalplayer-i18n.c
===================================================================
--- vdr-plugin-externalplayer-0.1.0.99.orig/externalplayer-i18n.c	2012-03-27 15:28:04.833843286 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,31 +0,0 @@
-/*
- * externalplayer-i18n.c: A plugin for the Video Disk Recorder
- *
- * See the README file for copyright information and how to reach the author.
- *
- * $Id$
- */
-
-#include "externalplayer-i18n.h"
-
-const tI18nPhrase ExternalplayerPhrases[] = {
-  { "External Players",
-    "Externe Abspielprogramme",
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-    "",// TODO
-  },
-  { NULL }
-  };
Index: vdr-plugin-externalplayer-0.1.0.99/externalplayer-i18n.h
===================================================================
--- vdr-plugin-externalplayer-0.1.0.99.orig/externalplayer-i18n.h	2012-03-27 15:28:08.903843288 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,16 +0,0 @@
-/*
- * externalplayer-i18n.h: A plugin for the Video Disk Recorder
- *
- * See the README file for copyright information and how to reach the author.
- *
- * $Id$
- */
-
-#ifndef _EXTERNALPLAYER_I18N__H
-#define _EXTERNALPLAYER_I18N__H
-
-#include <vdr/i18n.h>
-
-extern const tI18nPhrase ExternalplayerPhrases[];
-
-#endif //_EXTERNALPLAYER_I18N__H
Index: vdr-plugin-externalplayer-0.1.0.99/externalplayer.c
===================================================================
--- vdr-plugin-externalplayer-0.1.0.99.orig/externalplayer.c	2012-03-27 15:29:39.913843344 +0200
+++ vdr-plugin-externalplayer-0.1.0.99/externalplayer.c	2012-03-27 18:22:54.003849622 +0200
@@ -12,7 +12,6 @@
 #include <getopt.h>
 
 #include "externalplayer.h"
-#include "externalplayer-i18n.h"
 #include "externalplayer-player.h"
 #include "externalplayer-control.h"
 
@@ -62,7 +61,7 @@
 }
 
 bool cPluginExternalplayer::Start() {
-  RegisterI18n(ExternalplayerPhrases);
+//  RegisterI18n(ExternalplayerPhrases);
 
   if (configFilename == "") {
     configFilename += ConfigDirectory();
